sound/soc/sof/compress.c

```diff
@@ -46,15 +46,17 @@ void snd_sof_compr_init_elapsed_work(struct work_struct *work)
  */
 void snd_sof_compr_fragment_elapsed(struct snd_compr_stream *cstream)
 {
-	struct snd_soc_pcm_runtime *rtd = cstream->private_data;
-	struct snd_compr_runtime *crtd = cstream->runtime;
+	struct snd_soc_pcm_runtime *rtd;
+	struct snd_compr_runtime *crtd;
 	struct snd_soc_component *component;
 	struct snd_compr_tstamp *tstamp;
 	struct snd_sof_pcm *spcm;
 
 	if (!cstream)
 		return;
 
+	rtd = cstream->private_data;
+	crtd = cstream->runtime;
```

Comment:
If `snd_sof_compr_fragment_elapsed()` is called then it is guarantied that `stream->cstream` is not NULL, but it is true that it is not checked directly by the caller (`ipc3_period_elapsed()`). The decision is made based on the `spcm->pcm.compress` which if it is true then the stream is compressed, thus `stream->cstream` is not NULL.